src/media/js/addon/components/comm.js

```diff
@@ -52,3 +55,54 @@ export default class Note extends React.Component {
     );
   }
 }
+
+
+export class NoteSubmit extends React.Component {
+  static propTypes = {
+    submitNote: React.PropTypes.func.isRequired,
+    threadId: React.PropTypes.number.isRequired,
+    versionId: React.PropTypes.number.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isVisible: false,
+      text: ''
+    };
+  }
+
+  handleTextChange = e => {
+    this.setState({text: e.target.value});
+  }
+
+  toggle = () => {
+    this.setState({isVisible: !this.state.isVisible});
+  }
+
+  submitNote = e => {
+    e.preventDefault();
+    this.props.submitNote(this.props.threadId, this.props.versionId,
+                          this.state.text);
+  }
+
+  render() {
+    return (
+      <div class="note-submit">
+        <button onClick={this.toggle}>
+          {this.state.isVisible ? 'Cancel Reply' : 'Reply'}
+        </button>
+
+        {this.state.isVisible &&
+          <form>
+            <textarea onChange={this.handleTextChange}
```

Comment:
the callback handler isn't that bad, it just changes a value in an object. it'd be a microoptimization